github-actions/generate-doc/action.yml

```diff
@@ -0,0 +1,47 @@
+name: 'Generate Readme Doc'
+description: 'Generate Readme MD File'
+inputs:
+  GITHUB_TOKEN:
+    description: 'PAT used to write to the target repo'
+    required: true
+runs:
+  using: 'composite'
+  steps:
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.9'
+    - name: 'Check PIP Cache'
+      uses: actions/cache@v2
+      id: cache
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip-soar-app-ci
+    - name: 'Install Dependencies'
+      run: |
+        sudo apt-get update && sudo apt-get install -y tidy
+        wget https://github.com/jgm/pandoc/releases/download/2.16.2/pandoc-2.16.2-1-amd64.deb
+        sudo dpkg -i pandoc-2.16.2-1-amd64.deb
+        python -m pip install --upgrade pip
+        pip install -r ${{ github.action_path }}/requirements.txt
+      shell: bash
+    - name: Checkout App Repo
+      uses: actions/checkout@v3
+      with:
+        ref: ${{ github.ref }}
+    - name: 'Generate ReadMe Doc'
+      env:
+        GITHUB_TOKEN: ${{ inputs.GITHUB_TOKEN }}
+        APP_REPO: ${{ github.repository }}
+      run: |
+        export APP_REPO_DIR="$(pwd)"
+        cd ${{ github.action_path }}
+        cd ../start-release
+        python build_docs.py $APP_REPO_DIR True
+        cd $APP_REPO_DIR
+        git config user.name "splunk-soar-connectors-admin"
+        git config user.email "admin@splunksoar"
+        git add README.md
+        git commit -m "Update README.md"
+        git push
```

Comment:
what happens if there a README rebuild gets triggered but there are no changes? for example, if you change a field in the app JSON that doesn't affect the README. does that fail the job?

Reply:
So, whenever there is json file change of the app, this action will be triggered. https://github.com/phantomcyber/copy-repository-scripts/pull/37/files
If app json file not change, this will not be triggered, and no new README.md generated.
If app json file change which is related or not related with doc, generate doc will still run and replace README.md. 

Ex. I added field "mary_test": true and change "fips_compliant": false in smtp.json, when this file is pushed, generate doc will be triggered, new README.md will be generated and pushed to local branch.

Reply:
right but if you generate a new README and it's the same as the old one then there are no changes from git's perspective. so doing git add wouldn't do anything and git commit would be trying to create an empty commit (which fails by default)

ideally, we should only create the README commit if there were actual changes generated. otherwise, it's going to be noisy and hard to find when the README is actually changed

Reply:
Right, updated this part when there is README.md change.

Reply:
More testing have been done:

Update smtp.json with no README.md change:
commit: https://github.com/splunk-soar-connectors/smtp/commit/de52ac4e9e375373f8a72466c5d9cf7f4b68d72f
job run: https://github.com/splunk-soar-connectors/smtp/actions/runs/4209066948/jobs/7305737348

Update smtp.json with README.md change:
commit: https://github.com/splunk-soar-connectors/smtp/commit/a6e1930243ecc97ed666c6dc210596df6f54a3a8
job run: https://github.com/splunk-soar-connectors/smtp/actions/runs/4209192545/jobs/7305971587

Update readme.html with README.md change:
commit: https://github.com/splunk-soar-connectors/smtp/commit/8d836ef746121bd53b0714a50dae4285b4f6ee95
job run: https://github.com/splunk-soar-connectors/smtp/actions/runs/4209160602/jobs/7305906874